src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -141,9 +141,9 @@ internal TypeTable GetTypeTable()
             return retValue;
         }
 
-        private static T AdapterGetFirstOrDefaultMemberDelegate<T>(PSObject msjObj, MemberNamePredicate predicate) where T : PSMemberInfo
+        private static T AdapterGetFirstMemberOrDefaultDelegate<T>(PSObject msjObj, MemberNamePredicate predicate) where T : PSMemberInfo
         {
-            if (msjObj.IsDeserialized && typeof(PSPropertyInfo).IsAssignableFrom(typeof(T)))
+            if (msjObj.IsDeserialized && typeof(T).IsAssignableFrom(typeof(PSPropertyInfo)))
```

Comment:
When `T` is `PSMemberInfo`, returning `PSPropertyInfo` object is perfectly fine.